docs/en/operations/system-tables/information_schema.md

```diff
@@ -218,3 +226,62 @@ is_trigger_updatable:       NO
 is_trigger_deletable:       NO
 is_trigger_insertable_into: NO
 ```
+
+## KEY_COLUMN_USAGE (#key_column_usage)
+
+It was added for compatibility with third party tools such as Tableau Online. Contains only the primary keys columns read from [system.columns](../../operations/system-tables/columns.md).
+
+Columns:
+
+- `constraint_catalog` ([String](../../sql-reference/data-types/string.md)) — The name of the catalog to which the constraint belongs. This value is always `def`.
+- `constraint_schema` ([String](../../sql-reference/data-types/string.md)) — The name of the schema (database) to which the constraint belongs.
+- `constraint_name` ([Nullable](../../sql-reference/data-types/nullable.md)([String](../../sql-reference/data-types/string.md))) — The name of the constraint.
+- `table_catalog` ([String](../../sql-reference/data-types/string.md)) — The name of the catalog to which the table belongs. This value is always `def`.
+- `table_schema` ([String](../../sql-reference/data-types/string.md)) — The name of the schema (database) to which the table belongs.
+- `table_name` ([String](../../sql-reference/data-types/string.md)) — The name of the table that has the constraint.
+- `column_name` ([Nullable](../../sql-reference/data-types/nullable.md)([String](../../sql-reference/data-types/string.md))) — The name of the column that has the constraint.
+- `ordinal_position` ([UInt32](../../sql-reference/data-types/int-uint.md)) — The column's position within the constraint, not the column's position within the table. Column positions are numbered beginning with 1.
+- `position_in_unique_constraint` ([Nullable](../../sql-reference/data-types/nullable.md)([UInt32](../../sql-reference/data-types/int-uint.md))) — Always `NULL`.
+- `referenced_table_schema` ([Nullable](../../sql-reference/data-types/nullable.md)([String](../../sql-reference/data-types/string.md))) — The name of the schema referenced by the constraint. Always `NULL`.
+- `referenced_table_name` ([Nullable](../../sql-reference/data-types/nullable.md)([String](../../sql-reference/data-types/string.md))) — The name of the table referenced by the constraint. Always `NULL`.
+- `referenced_column_name` ([Nullable](../../sql-reference/data-types/nullable.md)([String](../../sql-reference/data-types/string.md))) — The name of the column referenced by the constraint. Always `NULL`.
+
+**Example**
+
+```sql
+CREATE TABLE test (i UInt32, s String) ENGINE MergeTree ORDER BY i;
```

Comment:
It is actually possible to define constraints for a table: https://clickhouse.com/docs/en/sql-reference/statements/create/table#constraints

Would you like to check if these are somehow exposed in the base view `system.columns`? If so, we should try to show them in `information_schema.key_column_usage` as well (right now, we don't). And if we show something, it would be great if the example (l. 252) could include a toy constraint.

Reply:
Does not look like it.

```sql
CREATE TABLE _test
(
    `i` UInt32,
    `j` UInt32,
    CONSTRAINT y CHECK j > 42
)
ENGINE = MergeTree
ORDER BY i

SELECT *
FROM system.columns
WHERE table = '_test'
FORMAT Vertical

Row 1:
──────
database:                default
table:                   _test
name:                    i
type:                    UInt32
position:                1
default_kind:            
default_expression:      
data_compressed_bytes:   0
data_uncompressed_bytes: 0
marks_bytes:             0
comment:                 
is_in_partition_key:     0
is_in_sorting_key:       1
is_in_primary_key:       1
is_in_sampling_key:      0
compression_codec:       
character_octet_length:  ᴺᵁᴸᴸ
numeric_precision:       32
numeric_precision_radix: 2
numeric_scale:           0
datetime_precision:      ᴺᵁᴸᴸ

Row 2:
──────
database:                default
table:                   _test
name:                    j
type:                    UInt32
position:                2
default_kind:            
default_expression:      
data_compressed_bytes:   0
data_uncompressed_bytes: 0
marks_bytes:             0
comment:                 
is_in_partition_key:     0
is_in_sorting_key:       0
is_in_primary_key:       0
is_in_sampling_key:      0
compression_codec:       
character_octet_length:  ᴺᵁᴸᴸ
numeric_precision:       32
numeric_precision_radix: 2
numeric_scale:           0
datetime_precision:      ᴺᵁᴸᴸ
```